tools/release/stage-release.ts

```diff
@@ -147,6 +151,52 @@ class StageReleaseTask extends BaseReleaseTask {
     writeFileSync(this.packageJsonPath, JSON.stringify(newPackageJson, null, 2) + '\n');
   }
 
+  /**
+   * Ensures that the Angular version placeholder has been correctly updated to support
+   * given Angular versions. The following rules apply:
+   *   `N.x.x` requires Angular `^N.0.0 || (N+1).0.0-0`
+   *   `N.0.0-x` requires Angular `^N.0.0-0 || (N+1).0.0-0`
+   */
+  private _verifyAngularPeerDependencyVersion(newVersion: Version) {
+    const currentVersionRange = this._getAngularVersionPlaceholderOrExit();
+    const isMajorWithPrerelease = newVersion.minor === 0 && newVersion.patch === 0 &&
+        newVersion.prereleaseLabel !== null;
+    const requiredRange = isMajorWithPrerelease ?
+      `^${newVersion.major}.0.0-0 || ^${newVersion.major + 1}.0.0-0` :
+      `^${newVersion.major}.0.0 || ^${newVersion.major + 1}.0.0-0`;
+
+    if (requiredRange !== currentVersionRange) {
+      console.error(chalk.red(`  ✘   Cannot stage release. The required Angular version range ` +
+        `is invalid. The version range should be: ${requiredRange}`));
+      console.error(chalk.red(`      Please manually update the version range ` +
+        `in: ${BAZEL_RELEASE_CONFIG_PATH}`));
+      return process.exit(1);
+    }
+  }
+
+  /**
+   * Gets the Angular version placeholder from the bazel release config. If
+   * the placeholder could not be found, the process will be terminated.
+   */
+  private _getAngularVersionPlaceholderOrExit(): string {
+    const bzlConfigPath = join(this.projectDir, BAZEL_RELEASE_CONFIG_PATH);
+    if (!existsSync(bzlConfigPath)) {
+      console.error(chalk.red(`  ✘   Cannot stage release. Could not find the file which sets ` +
+        `the Angular peerDependency placeholder value. Looked for: ${bzlConfigPath}`));
+      return process.exit(1);
+    }
+
+    const configFileContent = readFileSync(bzlConfigPath, 'utf8');
+    const matches = configFileContent.match(/ANGULAR_PACKAGE_VERSION = ["']([^"']+)/);
```

Comment:
I do not know of a better way to get the version placeholder from the Bazel `.bzl` file.

Reply:
You could make a `nodejs_binary` that takes in the version placeholder as an argument and writes it to stdout or a `genrule` that outputs the placeholder to a file.

Reply:
Yeah, that's what we are doing for the rollup globals check (the genrule). Though it would complicate things if a file needs to be built by Bazel whenever the staging script runs. i.e. would we just spawn a process to build it?

The `nodejs` binary one is more interesting, but unfortunately it would complicate things too. If you remember, we used Bazel initially for the release scripts, but the scripts ran in the Bazel bin directory..

Reply:
For the genrule, yeah, that's what I had in mind (spawn a child process).

Reply:
Yeah, that would work. I don't mind much either. Happy to do the genrule if you slightly prefer it more than the other options. I'm somehow preferring the current one since we basically just need static analysis and actually building seems unnecessary. It's a more robust idea though.